modules/_labs/select/react/lib/SelectMenu.tsx

```diff
@@ -1,18 +1,47 @@
-import * as React from 'react';
+import React, {useState, useEffect} from 'react';
+import {Rect} from '@popperjs/core';
+
 import {CSSObject, keyframes} from '@emotion/core';
 import {EmotionCanvasTheme, ErrorType, Themeable, styled} from '@workday/canvas-kit-react-common';
-import {colors, borderRadius, inputColors, spacingNumbers} from '@workday/canvas-kit-react-core';
+import {Placement, Popper, useCloseOnEscape} from '@workday/canvas-kit-react-popup';
+import {colors, borderRadius, inputColors} from '@workday/canvas-kit-react-core';
+
 import {SelectProps} from './Select';
+import {buttonBorderWidth} from './SelectBase';
 
 interface SelectMenuProps
   extends Themeable,
     React.HTMLAttributes<HTMLUListElement>,
     Pick<SelectProps, 'error'> {
+  /**
+   * The ref to the anchor button element. Required to anchor the menu.
+   */
+  buttonRef: React.RefObject<HTMLButtonElement>;
   /**
    * If true, enable animation on the SelectMenu.
    * @default true
    */
   isAnimated: boolean;
+  /**
+   * If true, automatically flip the SelectMenu to keep it visible if necessary (e.g., if the the SelectMenu would otherwise display below the visible area of the viewport).
+   * @default true
+   */
+  isAutoFlipped: boolean;
+  /**
+   * If true, focus the SelectMenu when it's shown. Set to false if you don't want to focus the SelectMenu automatically (for visual testing purposes, for example).
+   * @default true
+   */
+  isAutoFocused: boolean;
+  /**
+   * If true, flip the SelectMenu so it extends upwards from the button.
+   * @default false
+   */
+  isFlipped: boolean;
+  /**
+   * If true, hide the SelectMenu.
+   * @default false
+   */
+  isHidden: boolean;
```

Comment:
What is the difference between `isHidden` and `isHiding`?

Reply:
`isHiding` indicates the menu is animating (fading) out. `isHidden` indicates the menu is completely hidden.

Reply:
I see. Now we're getting into state machine territory. You can usually tell if you have a bunch of booleans to determine state. But we have a weird issue here. Not all permutations of `isHidden` and `isHiding` are valid. We only have 3 states, but there are 4 permutations.

For example,
|  | isHidden=true  | isHidden=false  |
|---|---|---|
| isHiding=true  | N/A | Currently hiding |
| isHiding=false  | Hidden | Not Hidden |

Also I noticed a lot of negative variable names. Example `isHidden` vs `isOpen`. It is generally considered a bad practice as negating a negative variable name results in a double-negative that is harder to understand: https://softwareengineering.stackexchange.com/questions/196830/boolean-method-naming-affirmative-vs-negative